subsys/bluetooth/host/l2cap.c

```diff
@@ -1062,7 +1062,13 @@ static void le_conn_req(struct bt_l2cap *l2cap, uint8_t ident,
 
 	/* Check if connection has minimum required security level */
 	if (!l2cap_check_security(conn, server)) {
-		rsp->result = sys_cpu_to_le16(BT_L2CAP_LE_ERR_AUTHENTICATION);
+		if (server->sec_level == BT_SECURITY_L2) {
```

Comment:
I think it would be better to have l2cap_check_security return the error code.

Also the error codes don't seem right. L2 -> ERR_ENCRYPTION, L3 -> ERR_AUTHENTICATION.